core/dbt/exceptions.py

```diff
@@ -257,6 +257,33 @@ def __reduce__(self):
         return (JSONValidationException, (self.typename, self.errors))
 
 
+class IncompatibleSchemaException(RuntimeException):
+    def __init__(self, expected: str, found: Optional[str]):
+        self.expected = expected
+        self.found = found
+        self.filename = 'input file'
+
+        super().__init__(self.get_message())
+
+    def add_filename(self, filename: str):
+        self.filename = filename
+        self.msg = self.get_message()
+
+    def get_message(self) -> str:
+        found_str = 'nothing'
+        if self.found is not None:
+            found_str = f'"{self.found}"'
+
+        msg = (
+            f'Expected a schema version of "{self.expected}" in '
+            f'{self.filename}, but found {found_str}'
```

Comment:
What do you think of offering a hint to users who might be initially confused about what `schema version` means?
```
  Expected a schema version of "https://schemas.getdbt.com/dbt/manifest/v1.json" in
.state/manifest.json, but found "https://schemas.getdbt.com/dbt/manifest/v2.json".
Are you running with a different version of dbt?
```